ilastik/applets/tracking/conservation/opConservationTracking.py

```diff
@@ -498,62 +525,96 @@ def stackExtraColumnWithZeros(array):
-        if self.resolvedMergersDict:
-            for timestep, results in self.resolvedMergersDict.items():
+            # Write merger results dictionary
+            resolvedMergersDict = self.ResolvedMergers.value
+            
+            if resolvedMergersDict:
                 mergerRes = {}
-                for key, result in results.items():
-                    mergerRes[key] = result
+                
+                for idx in mergersPerTimestep[timestep]:
+                    mergerRes[idx] = resolvedMergersDict[int(timestep)][idx]['newIds']
                     
-                events[str(timestep)]['res'] = mergerRes
+                events[timestep]['res'] = mergerRes
+                
         else:
             logger.info("Resolved Merger Dictionary not available. Please click on the Track button.")
                 
         return events
 
+    def _labelMergers(self, volume, time, offset):
+        """
+        Label volume mergers with correspoding IDs, using the plugin GMM fit
+        """
+        resolvedMergersDict = self.ResolvedMergers.value
+        
+        if time not in resolvedMergersDict:
+            return volume
+        
+        idxs = vigra.analysis.unique(volume)
+        
+        for idx in idxs: 
+            if idx in resolvedMergersDict[time]:
+                fits = resolvedMergersDict[time][idx]['fits']
+                newIds = resolvedMergersDict[time][idx]['newIds']
+                self.mergerResolverPlugin.updateLabelImage(volume, idx, fits, newIds, offset=offset)
+        
+        return volume               
+
     def _labelLineageIds(self, volume, time, onlyMergers=False):
         """
         Label the every object in the volume for the given time frame by the lineage ID it belongs to.
         If onlyMergers is True, then only those segments that were resolved from a merger are shown, everything else set to zero.
 
         :return: the relabeled volume, where 0 means background, 1 means false detection, and all higher numbers indicate lineages
         """
-        if self.hypotheses_graph:
-            indexMapping = np.zeros(np.amax(volume) + 1, dtype=volume.dtype)
+        hypothesesGraph = self.HypothesesGraph.value
+        
+        if not hypothesesGraph:
+            return np.zeros_like(volume) 
+        
+        resolvedMergersDict = self.ResolvedMergers.value
+
+        indexMapping = np.zeros(np.amax(volume) + 1, dtype=volume.dtype)
+        
+        idxs = vigra.analysis.unique(volume)
+        
+        # Reduce labels to the ones that contain mergers
+        if onlyMergers:
+            if resolvedMergersDict:
+                if time not in resolvedMergersDict:
+                    idxs = []
+                else:
+                    newIds = [nodeDict['newIds'] for _, nodeDict in resolvedMergersDict[time].items()]
+                    newIds = [id for ids in newIds for id in ids]
+                    idxs = [id for id in idxs if id in newIds]
```

Comment:
Style note: `id` is the name of a builtin function, so it's best not to use it as a variable name.